### PR TITLE
ROCm: Add Backend

### PR DESCRIPTION
**Pull request by crtrott**
Do NOT merge yet, this is to keep track of potential merge issues.

### CODE REVIEW

File: core/unit_test/Makefile [hunk 1]
```diff
@@ -209,7 +258,8 @@ endif
         OBJ_SERIAL += TestSerial_Team.o TestSerial_TeamScratch.o
         OBJ_SERIAL += TestSerial_TeamReductionScan.o
         OBJ_SERIAL += TestSerial_Other.o
-        OBJ_SERIAL += TestSerial_MDRange.o
+        #HCC_WORKAROUND
```
Comment (crtrott) [inline core/unit_test/Makefile]: We need to fix this really.

**Comment by crtrott** [comment]
Feel free to start reviewing though.

File: core/unit_test/Makefile (hunk 1 above)
Comment (hcedwar) [inline core/unit_test/Makefile]: fix before merge

File: core/unit_test/Makefile [hunk 2]
```diff
@@ -226,8 +276,8 @@ ifneq ($(KOKKOS_INTERNAL_USE_OPENMPTARGET), 1)
   OBJ_DEFAULT += TestDefaultDeviceType.o TestDefaultDeviceType_a.o TestDefaultDeviceType_b.o TestDefaultDeviceType_c.o TestDefaultDeviceType_d.o
 endif
 
-TARGETS += KokkosCore_UnitTest_Default
-TEST_TARGETS += test-default
+#TARGETS += KokkosCore_UnitTest_Default
+#TEST_TARGETS += test-default
```
Comment (hcedwar) [inline core/unit_test/Makefile]: Add back before merge

**Pull request merged**